driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -164,14 +164,35 @@ public RequestHandler(
     ConsistencyLevel consistency = statement.getConsistencyLevel();
     if (consistency == null) consistency = Statement.DEFAULT.getConsistencyLevel();
 
-    String statementType = "regular";
-    if (statement instanceof BatchStatement) statementType = "batch";
-    else if (statement instanceof PreparedStatement) statementType = "prepared";
+    String statementType = "";
+    String statementText = "";
+    int batchSize = 1;
+    if (statement instanceof BatchStatement) {
+      statementType = "batch";
+      batchSize = ((BatchStatement) statement).size();
+      StringBuilder statementTextBuilder = new StringBuilder(1000);
```

Comment:
Better to make `1000` a named constant instead of a magic number here (and in one other place). It's probably fine to keep it non-user-modifiable for now (`private static final int STATEMENT_MAX_LENGTH` on top of this file).